nemo/collections/asr/parts/submodules/multitask_decoding.py

```diff
@@ -0,0 +1,493 @@
+        return hypothesis
+
+    def decode_ids_to_tokens(self, tokens: List[int]) -> List[str]:
+        """
+        Implemented by subclass in order to decode a token id list into a token list.
+        A token list is the string representation of each token id.
+
+        Args:
+            tokens: List of int representing the token ids.
+
+        Returns:
+            A list of decoded tokens.
+        """
+        token_list = self.tokenizer.ids_to_tokens(tokens)
+        return token_list
+
+    def decode_tokens_to_lang(self, tokens: List[int]) -> str:
+        """
+        Compute the most likely language ID (LID) string given the tokens.
+
+        Args:
+            tokens: List of int representing the token ids.
+
+        Returns:
+            A decoded LID string.
+        """
+        lang = self.tokenizer.ids_to_lang(tokens)
+        return lang
+
+    def decode_ids_to_langs(self, tokens: List[int]) -> List[str]:
+        """
+        Decode a token id list into language ID (LID) list.
+
+        Args:
+            tokens: List of int representing the token ids.
+
+        Returns:
+            A list of decoded LIDS.
+        """
+        lang_list = self.tokenizer.ids_to_text_and_langs(tokens)
+        return lang_list
+
+    def decode_hypothesis(self, hypotheses_list: List[Hypothesis]) -> List[Union[Hypothesis, NBestHypotheses]]:
+        """
+        Decode a list of hypotheses into a list of strings.
+        Overrides the super() method optionally adding lang information
+
+        Args:
+            hypotheses_list: List of Hypothesis.
+
+        Returns:
+            A list of strings.
+        """
+        hypotheses = super().decode_hypothesis(hypotheses_list)
+        if self.compute_langs:
+            if isinstance(self.tokenizer, AggregateTokenizer):
+                for ind in range(len(hypotheses_list)):
+                    # Extract the integer encoded hypothesis
+                    prediction = hypotheses_list[ind].y_sequence
+
+                    if type(prediction) != list:
+                        prediction = prediction.tolist()
+
+                    hypotheses[ind].langs = self.decode_tokens_to_lang(prediction)
+                    hypotheses[ind].langs_chars = self.decode_ids_to_langs(prediction)
+            else:
+                logging.warning(
+                    "Ignoring request for lang output in hypotheses since the model does not use an aggregate tokenizer"
+                )
+
+        return hypotheses
+
+
+@dataclass
+class MultiTaskDecodingConfig:
+    strategy: str = "beam"
+
+    compute_hypothesis_token_set: bool = False
+
+    # preserve decoding alignments
+    preserve_alignments: Optional[bool] = None
+
+    # compute language IDs
+    compute_langs: bool = False
+
+    # greedy decoding config
+    # greedy: rnnt_greedy_decoding.GreedyBatchedRNNTInferConfig = field(
+    #     default_factory=rnnt_greedy_decoding.GreedyBatchedRNNTInferConfig
+    # )
```

Comment:
Remove/restore commented out code (rnnt probably doesn't belong here?)

Reply:
This is placeholder for @AlexGrinch to add greedy decoding strategy